opm/parser/eclipse/EclipseState/SimulationConfig/HysteresisOptions.cpp

```diff
@@ -0,0 +1,230 @@
+        case  8: return Model::JargonHysteresisNonWettingDrainageWetting;
+        case  9: return Model::JargonHysteresisNonWettingImbibitionWetting;
+        default:
+            throw std::invalid_argument( "Hysteresis model must be an integer [-1, 9]" );
+    }         
+}
+
+int readHysteresisApply( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) ) return 0;
+
+    const auto& key = uppercase( item.getTrimmedString( 0 ) );
+
+    if( key == "BOTH" ) return 0;
+    if( key == "PC" ) return 1;
+    if( key == "KR" ) return 2;
+
+    throw std::invalid_argument( "Limiting hysteresis flag must be 'PC', 'KR' or 'BOTH'" );
+}
+
+HysteresisOptions::CapillaryPressureScanningCurveShape readHysteresisCurveShape( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) )
+        return HysteresisOptions::CapillaryPressureScanningCurveShape::RETR;
+
+    const auto& key = uppercase( item.getTrimmedString( 0 ) );
+    if( key == "RETR" )
+        return HysteresisOptions::CapillaryPressureScanningCurveShape::RETR;
+
+    if( key == "NEW" )
+        return HysteresisOptions::CapillaryPressureScanningCurveShape::RETR;
+
+    throw std::invalid_argument( "Flag controlling the shape of capillary pressure curves on retraversal must be 'RETR' or 'NEW'" );
+}
+
+bool readHysteresisCorrectImbibition( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) ) return false;
+
+    const auto& key = uppercase( item.getTrimmedString( 0 ) );
+    if( key == "DRAIN" ) return false;
+    if( key == "BOTH" ) return true;
+
+    throw std::invalid_argument( "Flag controlling fluid mobility correction application on imbibition curves must be 'DRAIN' or 'BOTH'" );
+}
+
+Phase readHysteresisWettingPhase( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) ) return Phase::OIL;
+
+    const auto& key = uppercase( item.getTrimmedString( 0 ) );
+    if( key == "OIL" ) return Phase::OIL;
+    if( key == "GAS" ) return Phase::GAS;
+
+    throw std::invalid_argument( "Hysteresis Wetting phase must be 'OIL' or 'GAS'" );
+}
+
+bool yesno( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) ) return false;
+
+    const auto& key = item.get< std::string >( 0 );
+    if( key.front() == 'Y' || key.front() == 'y' ) return true;
+    if( key.front() == 'N' || key.front() == 'n' ) return true;
+
+    throw std::invalid_argument( "Hysteresis boolean flag must be 'YES' or 'NO'" );
+}
+
+}
+
+HysteresisOptions::HysteresisOptions( const Deck& deck ) {
+    if( !deck.hasKeyword( "EHYSTR" ) ) return;
+
+    const auto& record = deck.getKeyword( "EHYSTR" ).getRecord( 0 );
+
+    this->enabled = true;
+    this->curvature = record.getItem( 0 ).get< double >( 0 );
+    this->hys_model = readHysteresisModel( record.getItem( 1 ).get< int >( 0 ) );
+    this->curvature_killough = record.getItem( 2 ).get< double >( 0 );
+    this->modification_killough = record.getItem( 3 ).get< double >( 0 );
```

Comment:
I suppose that the `this->` is superfluous. Also, isn't the opm-parser convention for private attributes `m_camelCase`?